libs/core/src/main/java/org/opensearch/core/common/bytes/AbstractBytesReference.java

```diff
@@ -313,7 +313,7 @@ public int available() {
         }
 
         @Override
-        protected void ensureCanReadBytes(int bytesToRead) throws EOFException {
+        public void ensureCanReadBytes(int bytesToRead) throws EOFException {
```

Comment:
moving classes around may require changing access modifiers but comes with a security risk. This one seems harmless however we can move most the of consumers of this classes to `lib:core` from server. I also see its unit tests are still in server. 
If we are not intending to move unit tests and other consumers of this class here, then we should create a list of TODOs, to change them back to their original access modifiers so that we don't miss them later.

Reply:
Yeah @rishabhmaurya, I actually like the idea of refactoring these classes as well. The reason I held off is because one of those inheritors, BufferedChecksumStreamInput, is inside the `o.o.index.translog` namespace and we may want to consider having this in a dedicated `:libs:opensearch-translog` library instead of core so we don't set a precedent of having translog logic in the core library.

Reply:
Oh by the way.. this PR is going to get closed in favor of https://github.com/opensearch-project/OpenSearch/pull/8157 so have a look at that one instead if you wouldn't mind.